chain/proofs/proofs.go

```diff
@@ -0,0 +1,79 @@
+package proofs
+
+import (
+	"math/bits"
+
+	"github.com/ipfs/go-cid"
+
+	"github.com/filecoin-project/go-commp-utils/nonffi"
+	"github.com/filecoin-project/go-commp-utils/zerocomm"
+	"github.com/filecoin-project/go-state-types/abi"
+)
+
+// GenerateUnsealedCID generates the UnsealedCID for a sector of size determined by the proofType
+// containing pieces of the specified sizes. Where there is not one piece the size of the sector, or
+// zero pieces (in which case one piece can be used), it fills in the remaining space with pieces of
+// the minimum size required to pad the sector.
+func GenerateUnsealedCID(proofType abi.RegisteredSealProof, pieces []abi.PieceInfo) (cid.Cid, error) {
```

Comment:
this originally comes from ./storage/sealer/ffiwrapper/sealer_cgo.go but gets further minimised when https://github.com/filecoin-project/lotus/pull/12345 gets pulled in after this PR